launcher/ui/dialogs/ResourceDownloadDialog.cpp

```diff
@@ -155,24 +159,10 @@ void ResourceDownloadDialog::addResource(ModPlatform::IndexedPack& pack, ModPlat
     m_buttons.button(QDialogButtonBox::Ok)->setEnabled(!m_selected.isEmpty());
 }
 
-static ModPlatform::IndexedVersion& getVersionWithID(ModPlatform::IndexedPack& pack, QVariant id)
-{
-    Q_ASSERT(pack.versionsLoaded);
-    auto it = std::find_if(pack.versions.begin(), pack.versions.end(), [id](auto const& v) { return v.fileId == id; });
-    Q_ASSERT(it != pack.versions.end());
-    return *it;
-}
-
 void ResourceDownloadDialog::removeResource(ModPlatform::IndexedPack& pack, ModPlatform::IndexedVersion& ver)
 {
-    if (auto selected_task_it = m_selected.find(pack.name); selected_task_it != m_selected.end()) {
-        auto selected_task = *selected_task_it;
-        auto old_version_id = selected_task->getVersionID();
-
-        // If the new and old version IDs don't match, search for the old one and deselect it.
-        if (ver.fileId != old_version_id)
-            getVersionWithID(pack, old_version_id).is_currently_selected = false;
-    }
+    dynamic_cast<ResourcePage*>(m_container->getPage(Modrinth::id()))->removeResourceFromPage(pack.name);
+    dynamic_cast<ResourcePage*>(m_container->getPage(Flame::id()))->removeResourceFromPage(pack.name);
```

Comment:
```suggestion
    static_cast<ResourcePage*>(m_container->getPage(Modrinth::id()))->removeResourceFromPage(pack.name);
    static_cast<ResourcePage*>(m_container->getPage(Flame::id()))->removeResourceFromPage(pack.name);
```

, though i'd suggest iterating over all the pages in `m_container`, so that this doesn't need to change when adding a new page in the future.

Reply:
I added a method on m_container to get all pages and iterate over it.